### PR TITLE
Add log category to formatted output.

### LINKED ISSUE #39: Display the category with the logger output
It would be nice to be able to quickly see what category a log message is related to from the log output.

### PR DESCRIPTION
**Pull request by karnkaul**
Ignore genesis.log.
Auto-format modified files.

Sample output:
```
[I][T0] [logger] logging to file: genesis.log [2023-08-06 22:17:12] [instance.cpp]
[I][T0] [general] Hello, world! [2023-08-06 22:17:12] [main.cpp]
```

Close #39

### CODE REVIEW

**Comment by karnkaul** [comment]
I've split the changes into two commits: the first one keeps the logic unchanged and applies formatting / gitignore changes. The second commit implements the feature.

File: engine/src/logger/instance.cpp [hunk 1]
```diff
@@ -12,277 +12,323 @@
 	#include <Windows.h>
```
Comment (Rinzii) [inline engine/src/logger/instance.cpp]: Prefer the "system/win32/windowsHeader.hpp" include in genesis over  the "Windows.h" include as it is far more light weight and removes the vast majority of unnecessary bloat brought in by the winapi.

File: engine/src/logger/instance.cpp [hunk 2]
```diff
@@ -12,277 +12,323 @@
...
-			return false;
+namespace gen::logger
+{
+	namespace
+	{
+		namespace fs = std::filesystem;
+
+		void append_timestamp(std::string & out, Clock::time_point const & timestamp, Timestamp const mode)
+		{
+			static auto s_mutex{std::mutex{}};
```
Comment (Rinzii) [inline engine/src/logger/instance.cpp:23]: Could you explain how you are avoiding the issues that come with std::gmtime here? Will the mutex be enough to ensure we don't have issues with the threads?

**Review by Rinzii — CHANGES_REQUESTED** [review]
I could not see really any major issues with the majority of the code, but there where a few small issues.

File: engine/src/logger/instance.cpp (hunk 2 above)
Comment (karnkaul) [reply engine/src/logger/instance.cpp:23]: `std::gmtime` and `std::localtime` may return a pointer to a shared an internal `static tm` instance, which is what makes it unsafe to use across multiple threads. We copy that returned `tm const*` by dereferencing it under a mutex lock, thus avoiding any data races.

**Review by Rinzii — APPROVED** [review]

**Pull request merged**